commit f39941fdcad6d1192b9d3ec612fa5ba79911803f (refs/changes/69/1269/2)
Author: Trent Mick <trentm@gmail.com>
Date:   2017-01-17T18:30:29-08:00 (2 years, 9 months ago)
    
    HEAD-2342 sdc-healthcheck's cnapi check doesn't like multiple headnodes
    Reviewed by: Josh Wilsdon <josh@wilsdon.ca>
    Approved by: Josh Wilsdon <josh@wilsdon.ca>

diff --git a/tools/bin/sdc-healthcheck b/tools/bin/sdc-healthcheck
index c4727a6f..72f0274a 100755
--- a/tools/bin/sdc-healthcheck
+++ b/tools/bin/sdc-healthcheck
@@ -81,9 +81,9 @@ function check_end_to_end()
       [[ $CLOUDAPI_STATUS -eq 0 ]] && return
       ;;
     cnapi)
-      local got_name=`sdc-cnapi /servers?headnode=true 2>/dev/null | \
-          json -H -a hostname`
-      [[ "$got_name" == "$(hostname)" ]] && return
+      # Allow for multiple headnodes: one of them should be this hostname.
+      sdc-cnapi /servers?headnode=true 2>/dev/null | json -H -a hostname \
+        | grep "^$(hostname)$" >/dev/null && return
       ;;
     fwapi)
       local got_status=`sdc-fwapi /rules 2>/dev/null | \
